src/varity/vcf_to_hgvs.clj

```diff
@@ -29,16 +29,38 @@
 (defn- coding-dna-ref-gene? [rg]
   (some? (re-matches #"(NM_|ENST)\d+(\.\d+)?" (:name rg))))
 
+(defn- insertion?
+  [del ins]
+  (and (zero? (count del)) (pos? (count ins))))
+
+(defn- var->start-end-cds-coord
+  [{:keys [pos ref alt]} {:keys [strand cds-start cds-end] :as rg}]
+  (let [[del ins offset] (diff-bases ref alt)
+        pos (+ pos offset)
+        pos (if (insertion? del ins)
+              (cond
+                (= strand :forward) (if (<= cds-end pos)
+                                      pos
+                                      (dec pos))
+                (= strand :reverse) (if (>= cds-start pos)
+                                      (dec pos)
+                                      pos))
+              pos)
+        ndel (count del)
+        end-pos (+ pos (if (zero? ndel) 0 (dec (count del))))
+        start-cds-coord (rg/cds-coord pos rg)
+        end-cds-coord (rg/cds-coord end-pos rg)]
+    {:start-cds-coord start-cds-coord
+     :end-cds-coord end-cds-coord}))
+
 (defn select-variant
   [var seq-rdr rg & {:keys [three-prime-rule]}]
   (if-let [nvar (normalize-variant var seq-rdr rg)]
-    (let [var-start-cds-coord (rg/cds-coord (:pos var) rg)
-          var-end-cds-coord (rg/cds-coord (+ (:pos var) (max (count (:ref var)) (count (:alt var)))) rg)
```

Comment:
When `ref: GATG`, `alt: G` and cds-start position is at `A`, first `G` position is `c.-1`.
So region of start-cds-coord is not the same as end-cds-region.